cmake/apple/generate_icons_apple.sh.in

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Recolor icon
+echo "Recoloring vector icon..."
+case "$1" in
+	"-r" ) # Release
+	sed 's/#780116/#249a56/g; s/#c51306/#50d99b/g' ../branding/icon.svg > recolored-icon.svg
+	sed 's/#780116/#249a56/g; s/#c51306/#50d99b/g' ../branding/splash.svg > recolored-splash.svg
+	;;
+	"-b" | "-rc" ) # Beta/Release Candidate
+	sed 's/#780116/#0267c1/g; s/#c51306/#09a9d9/g' ../branding/icon.svg > recolored-icon.svg
+	sed 's/#780116/#0267c1/g; s/#c51306/#09a9d9/g' ../branding/splash.svg > recolored-splash.svg
+	;;
+	"-c" | "-n" ) # Canary/Nightly
+	sed 's/#780116/#ffa40f/g; s/#c51306/#fad200/g' ../branding/icon.svg > recolored-icon.svg
+	sed 's/#780116/#ffa40f/g; s/#c51306/#fad200/g' ../branding/splash.svg > recolored-splash.svg
+	;;
+	* ) # No type specified, leave it be
+	sed '' ../branding/icon.svg > recolored-icon.svg
+	sed '' ../branding/splash.svg > recolored-splash.svg
+	;;
+esac
```

Comment:
I'm reviewing the scripts and I see many redundancies because platform-agnostic tasks (like the icon and the splash) are being run identically on all platforms.  I recommend we consolidate them.  Here's a visual diff.

![image](https://user-images.githubusercontent.com/6345473/64634122-8f260980-d3ca-11e9-87dd-0d014cf59f7a.png)

@iansannar would you consider moving these into one script with some platform detection?  This would also make converting it to `cmake` a bit smaller of a task as it'd be one script to convert instead of three.

```bash
if [ -n "@LMMS_BUILD_APPLE@" ]; then
  # apple-specific logic
elif [ -n "@ LMMS_BUILD_WIN32@" ]; then
  # windows-specific logic (Note: MSVC won't ever get here because bash is missing)
else
  # fallback to linux-specific logic
fi
```

Reply:
That's a good idea. I'll wait until I am on my own computer to do something like that though; one commit per file change through a web browser is getting messy.